dependency/client_set.go

```diff
@@ -205,15 +205,14 @@ func (c *ClientSet) CreateVaultClient(i *CreateVaultClientInput) error {
 		}
 
 		// Custom CA certificate
-		if i.SSLCACert != "" {
-			cacert, err := ioutil.ReadFile(i.SSLCACert)
-			if err != nil {
-				return fmt.Errorf("client set: vault: %s", err)
+		if i.SSLCACert != "" || i.SSLCAPath != "" {
+			rootConfig := &rootcerts.Config{
+				CAFile: i.SSLCACert,
+				CAPath: i.SSLCAPath,
+			}
+			if err := rootcerts.ConfigureTLS(&tlsConfig, rootConfig); err != nil {
+				return fmt.Errorf("client set: vault configuring TLS failed: %s", err)
 			}
```

Comment:
@sethvargo I am not sure if you want to change it but you can see how Nomad creates a Vault client. It just creates a default vault client config and then builds the TLS struct and calls this: https://github.com/hashicorp/vault/blob/master/api/client.go#L108

Makes it so that all the config parsing happens in Vault api/ and not the caller